### PR TITLE
Translate List.Exists into a store query

### PR DESCRIPTION
**Pull request by ilmax**
**Please check if the PR fulfills these requirements**

- [x] The code builds and tests pass (verified by our automated build checks)
- [x] Commit messages follow this format
```
    Summary of the changes
    - Detail 1
    - Detail 2

    Fixes #bugnumber
```
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Code meets the expectations our engineering guidelines. https://github.com/aspnet/Home/wiki/Engineering-guidelines.

Please review the guidelines for CONTRIBUTING.md for more details.

### CODE REVIEW

File: src/EFCore/Query/ExpressionVisitors/Internal/ExistsToAnyRewritingExpressionVisitor.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        {
+            if (methodCallExpression.Arguments.Count == 1 &&
+                methodCallExpression.Method.Name == nameof(List<int>.Exists) &&
+                methodCallExpression.Method.DeclaringType.IsGenericType &&
+                methodCallExpression.Method.DeclaringType.GetGenericTypeDefinition() == typeof(List<>))
+            {
+                var genericTypeParameters = methodCallExpression.Method.DeclaringType.GetGenericArguments();
+                if (genericTypeParameters.Length == 1 && methodCallExpression.Arguments[0] is LambdaExpression actualLambdaExpression)
+                {
+                    var innerListType = methodCallExpression.Method.DeclaringType.GetGenericArguments()[0];
```
Comment (smitpatel) [inline src/EFCore/Query/ExpressionVisitors/Internal/ExistsToAnyRewritingExpressionVisitor.cs]: Can use `genericTypeParameters`

Comment (ilmax) [reply src/EFCore/Query/ExpressionVisitors/Internal/ExistsToAnyRewritingExpressionVisitor.cs]: Will update later today or tomorrow evening CET time

Comment (smitpatel) [reply src/EFCore/Query/ExpressionVisitors/Internal/ExistsToAnyRewritingExpressionVisitor.cs]: @ilmax - Its alright I will update while merging.

Comment (ilmax) [reply src/EFCore/Query/ExpressionVisitors/Internal/ExistsToAnyRewritingExpressionVisitor.cs]: @smitpatel  I can do it now :smirk:

**Comment by ilmax** [comment]
@smitpatel just pushed, would you like me to rebase this commit and squash it into the first one?

**Comment by smitpatel** [comment]
Merged via 2fdd114c6efea5b4283adb129d482526d56b64e6

Thanks for contribution @ilmax

**Pull request closed without merging**